tuplex/core/include/Partition.h

```diff
@@ -110,6 +111,24 @@ namespace tuplex {
             setNumRows(0);
         }
 
+        explicit Partition(Partition* part) :
```

Comment:
nullptr check? what is this constructor doing? There are reasons why we don't have a copy constructor. I.e., partition should be not copyable.

Reply:
yes, I totally agree it shouldn't be copyable. The reason I need this is the need to split a partition into two small partitions when we take both top and bottom from the same partition. E.g.  `[1, 2, 3, ..., 4, 5, 6].take(top=2, bottom=2)` -> `{[1,2], [5,6]}`. I would love some suggestion on how to properly split here.

Reply:
just create two new partitions. That's anyways a good idea, because think of the issues that will happen for the following snippet if you overwrite the original partitions:

```
ds = c.parallelize([1, 2, 3, 4, 5, ...., 98, 99, 100])

ds.take(top=3, bottom=3) # <-- data lost/corrupted in original partitions?

ds.collect() # <-- should return everything.
```

Reply:
Yes, I can do it in executor logic.

Question: In the above case, would the wrapper `Partition` be shared between both take and collect? If that is the case, `take()` currectly (before this change) just change the num element in `Partition`, wouldn't that also cause the collect to miss some element?